e2e/specs/record-drug-orders.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { expect } from '@playwright/test';
+import { type Visit } from '@openmrs/esm-framework';
+import { generateRandomPatient, deletePatient, type Patient, startVisit, endVisit } from '../commands';
+import { test } from '../core';
+import { MedicationsPage } from '../pages';
+
+let patient: Patient;
+let visit: Visit;
+
+test.beforeEach(async ({ api }) => {
+  patient = await generateRandomPatient(api);
+  visit = await startVisit(api, patient.uuid);
+});
+
+test('Order a drug', async ({ page, api }) => {
+  const medicationsPage = new MedicationsPage(page);
+
+  await test.step('When I visit the medications page', async () => {
+    await medicationsPage.goTo(patient.uuid);
+  });
+
+  await test.step('And I click on the `Record active medications` link', async () => {
+    await medicationsPage.page.getByText(/record active medications/i).click();
+  });
+
+  await test.step('And I record a medication', async () => {
+    await expect(medicationsPage.page.getByText(/add drug order/i)).toBeVisible();
+    await medicationsPage.page.getByRole('searchbox', { name: /search for a drug or orderset/i }).fill('aspirin');
+    await expect(medicationsPage.page.getByText(/6 results for "aspirin"/)).toBeVisible();
+    await medicationsPage.page
+      .getByRole('listitem')
+      .filter({ hasText: /aspirin 81mg — 81mg — tablet/i })
+      .getByRole('button', { name: /add to basket/i })
+      .click();
+
+    await expect(medicationsPage.page.getByRole('heading', { name: 'Drug orders (1)' })).toBeVisible();
+    await medicationsPage.page
+      .getByRole('listitem')
+      .filter({ hasText: /incomplete/i })
+      .click();
+
+    await expect(medicationsPage.page.getByRole('button', { name: /back to order basket/i })).toBeVisible();
+    await medicationsPage.page.getByLabel(/^dose$/i).fill('1');
+    await medicationsPage.page.getByLabel(/^duration$/i).fill('3');
+    await medicationsPage.page.getByPlaceholder(/route/i).click();
+    await medicationsPage.page.getByText('Oral', { exact: true }).click();
+    await medicationsPage.page.getByPlaceholder(/frequency/i).click();
+    await medicationsPage.page.getByText('Once daily', { exact: true }).click();
+    await medicationsPage.page.getByLabel(/indication/i).fill('Headache');
+    await medicationsPage.page.getByRole('button', { name: /save order/i }).click();
+    await medicationsPage.page.getByRole('button', { name: /sign and close/i }).click();
+  });
+
+  await test.step('Then I should see a success notification', async () => {
+    await expect(medicationsPage.page.getByText(/placed order for aspirin/i)).toBeVisible();
+  });
```

Comment:
Shall we add another step to verify the posted data renders on the page properly?

Reply:
I agree. Unfortunately, I'm getting this error when running the spec locally:

![CleanShot 2023-11-23 at 1  28 06@2x](https://github.com/openmrs/openmrs-esm-patient-chart/assets/8509731/302cfd19-54d5-4a82-ba91-e1e98363bba1)

This suggests that common lib isn't seeing the workspace registration when running a local server. It's an annoying bug and the easiest solution is probably adding that expectation after this work gets merged.